dot/core/service.go

```diff
@@ -249,6 +250,13 @@ func (s *Service) handleBlock(block *types.Block, state *rtstorage.TrieState) er
 }
 
 func (s *Service) handleCodeSubstitution(hash common.Hash, state *rtstorage.TrieState) error {
+	return s._handleCodeSubstitution(hash, state, wasmer.NewInstance)
+}
+func (s *Service) _handleCodeSubstitution(
```

Comment:
I think we don't need `_` for a helper or private functions, you could split the `_handleCodeSubstitution` into `hasCodeSubstitue` and `substitute` then you call them sequentially inside `handleCodeSubstitution` like:

```go
func (s *Service) handleCodeSubstitution ... {
    if hasCodeSubstitue(hash) {
        applySubstitute(hash, state, instance)
    }
}
```

Reply:
what do you think @timwu20 @qdm12 ?

Reply:
@EclesioMeloJunior updated!